### PR TITLE
Extend toc in sidebar indentation

### PR DESCRIPTION
**Pull request by s-weigand**
Since for some auto-generated documentation, only having indentation till `navigation_depth==4` might lead to problems ([see](https://github.com/readthedocs/sphinx_rtd_theme/issues/540#issuecomment-515341625)), I followed @agjohnson 's [comment](https://github.com/readthedocs/sphinx_rtd_theme/issues/540#issuecomment-516114657) and implemented indentation until `navigation_depth==10`.

**What this PR does:**
* Indentation for elements of the toc in sidebar up to `navigation_depth==10`
* Fontsize shrinking only for elements of levels-{2,3,4}, as before, but with a placeholder selector.
* Generalization of some properties  for elements of the toc in sidebar with placeholder selectors and for loop.

### Before
before

### After
after

### CODE REVIEW

**Comment by eine** [comment]
ping @Blendify @ericholscher @agjohnson

File: .gitignore [hunk 1]
```diff
@@ -21,5 +21,6 @@ sphinx_rtd_theme/static/fonts/Lato/
 sphinx_rtd_theme/static/fonts/RobotoSlab/
 .python-version
 .node-version
+.vscode
```
Comment (Blendify) [inline .gitignore]: Unrelated change, please remove

Comment (s-weigand) [reply .gitignore]: Done

**Comment by Alexdara** [comment]
@s-weigand  
I can not update my python lib thought
` pip install git+https://github.com/readthedocs/sphinx_rtd_theme.git@master`
it returns：
`Installing collected packages: sphinx-rtd-theme
  Attempting uninstall: sphinx-rtd-theme
    Found existing installation: sphinx-rtd-theme 0.4.3
    Uninstalling sphinx-rtd-theme-0.4.3:
      Successfully uninstalled sphinx-rtd-theme-0.4.3
    Running setup.py install for sphinx-rtd-theme ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\program files\python37\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"'; __file__='"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\TOH-WZ~1\AppData\Local\Temp\pip-record-uh7x79kq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\program files\python37\Include\sphinx-rtd-theme'
         cwd: C:\Users\TOH-WZ~1\AppData\Local\Temp\pip-req-build-60liz4ry\
    Complete output (4 lines):
    running install
    running build
    running build_py
    error: [WinError 2] 系统找不到指定的文件。
    ----------------------------------------
  Rolling back uninstall of sphinx-rtd-theme
  Moving to c:\program files\python37\lib\site-packages\sphinx_rtd_theme-0.4.3.dist-info\
   from c:\program files\python37\lib\site-packages\~phinx_rtd_theme-0.4.3.dist-info
  Moving to c:\program files\python37\lib\site-packages\sphinx_rtd_theme\
   from c:\program files\python37\lib\site-packages\~phinx_rtd_theme
ERROR: Command errored out with exit status 1: 'c:\program files\python37\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"'; __file__='"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\TOH-WZ~1\AppData\Local\Temp\pip-record-uh7x79kq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\program files\python37\Include\sphinx-rtd-theme' Check the logs for full command output.`

Are there any suggestions？Thanks a lot.

**Comment by Alexdara** [comment]
> @s-weigand
> I can not update my python lib thought
> ` pip install git+https://github.com/readthedocs/sphinx_rtd_theme.git@master`
> it returns：
> `Installing collected packages: sphinx-rtd-theme Attempting uninstall: sphinx-rtd-theme Found existing installation: sphinx-rtd-theme 0.4.3 Uninstalling sphinx-rtd-theme-0.4.3: Successfully uninstalled sphinx-rtd-theme-0.4.3 Running setup.py install for sphinx-rtd-theme ... error ERROR: Command errored out with exit status 1: command: 'c:\program files\python37\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"'; __file__='"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\TOH-WZ~1\AppData\Local\Temp\pip-record-uh7x79kq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\program files\python37\Include\sphinx-rtd-theme' cwd: C:\Users\TOH-WZ~1\AppData\Local\Temp\pip-req-build-60liz4ry\ Complete output (4 lines): running install running build running build_py error: [WinError 2] The system cannot find the file specified. ---------------------------------------- Rolling back uninstall of sphinx-rtd-theme Moving to c:\program files\python37\lib\site-packages\sphinx_rtd_theme-0.4.3.dist-info\ from c:\program files\python37\lib\site-packages\~phinx_rtd_theme-0.4.3.dist-info Moving to c:\program files\python37\lib\site-packages\sphinx_rtd_theme\ from c:\program files\python37\lib\site-packages\~phinx_rtd_theme ERROR: Command errored out with exit status 1: 'c:\program files\python37\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"'; __file__='"'"'C:\\Users\\TOH-WZ~1\\AppData\\Local\\Temp\\pip-req-build-60liz4ry\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\TOH-WZ~1\AppData\Local\Temp\pip-record-uh7x79kq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\program files\python37\Include\sphinx-rtd-theme' Check the logs for full command output.`
> 
> Are there any suggestions？Thanks a lot.

Solved this by use 
`pip install -e git+https://github.com/readthedocs/sphinx_rtd_theme.git@master#egg=sphinx-rtd-theme`

**Comment by s-weigand** [comment]
I think that the problem with directly installing it from from git, in none dev mode, is that the setup needs a local installation of webpack.
See:
https://github.com/readthedocs/sphinx_rtd_theme/blob/ca015a57042713ebd7f39563cdb01fab5b604051/setup.py#L16

**Comment by Alexdara** [comment]
> I think that the problem with directly installing it from from git, in none dev mode, is that the setup needs a local installation of webpack.
> See:
> https://github.com/readthedocs/sphinx_rtd_theme/blob/ca015a57042713ebd7f39563cdb01fab5b604051/setup.py#L16-L24



> I think that the problem with directly installing it from from git, in none dev mode, is that the setup needs a local installation of webpack.
> See:
> https://github.com/readthedocs/sphinx_rtd_theme/blob/ca015a57042713ebd7f39563cdb01fab5b604051/setup.py#L16-L24

After install webpack,still returns error as above.
